src/minio/core/minio_client.py

```diff
@@ -81,3 +86,213 @@ async def test_connection(self) -> bool:
+
+        Raises:
+            MinIOManagerError: If listing buckets fails.
+        """
+        try:
+            async with self._get_client() as client:
+                response = await client.list_buckets()
+                buckets = [bucket["Name"] for bucket in response.get("Buckets", [])]
+                logger.info(f"Listed {len(buckets)} buckets")
+                return buckets
+        except Exception as e:
+            logger.error(f"Failed to list buckets: {e}")
+            raise MinIOManagerError(f"Bucket listing failed: {e}") from e
+
+    async def delete_bucket(self, bucket_name: str) -> bool:
+        """
+        Deletes a bucket and all objects within it.
+
+        This operation first lists and deletes all objects in the bucket before
+        deleting the bucket itself.
+
+        Args:
+            bucket_name: The name of the bucket to delete.
+
+        Returns:
+            True if the bucket and its contents were successfully deleted.
+
+        Raises:
+            BucketOperationError: If the bucket deletion fails.
+        """
+        try:
+            async with self._get_client() as client:
+                # Always delete all objects first
+                paginator = client.get_paginator("list_objects_v2")
+                async for page in paginator.paginate(Bucket=bucket_name):
+                    if "Contents" in page:
+                        objects = [{"Key": obj["Key"]} for obj in page["Contents"]]
+                        await client.delete_objects(
+                            Bucket=bucket_name, Delete={"Objects": objects}
+                        )
+
+                await client.delete_bucket(Bucket=bucket_name)
+                logger.info(f"Deleted bucket: {bucket_name}")
+                return True
+        except Exception as e:
+            logger.error(f"Failed to delete bucket {bucket_name}: {e}")
+            raise BucketOperationError(f"Bucket deletion failed: {e}") from e
+
+    async def put_object(self, bucket_name: str, key: str, body: bytes) -> bool:
+        """
+        Uploads an object to a specified bucket.
+
+        Args:
+            bucket_name: The name of the target bucket.
+            key: The object key (i.e., its name/path within the bucket).
+            body: The content of the object as bytes.
+
+        Returns:
+            True if the object was uploaded successfully.
+
+        Raises:
+            MinIOManagerError: If the upload operation fails.
+        """
+        try:
+            async with self._get_client() as client:
+                await client.put_object(Bucket=bucket_name, Key=key, Body=body)
+                logger.info(f"Put object {key} in bucket {bucket_name}")
+                return True
+        except Exception as e:
+            logger.error(f"Failed to put object {key} in bucket {bucket_name}: {e}")
+            raise MinIOManagerError(f"Object put failed: {e}") from e
```

Comment:
What are you thinking the use case for this might be? It's not compatible with the CTS for example. Is the server going to be sitting between the user and Minio like the blobstore?

Reply:
What if the body is a metagenome?

Reply:
> put a '.keep' file to persist the path in MinIO.

I still don't really understand why this is necessary. 

Otherwise sounds good other than I would add docs saying this should only be used for small objects

Reply:
I would need some helper function to create a file.

Reply:
Added a note in the docstring.

Reply:
> I would need some helper function to create a file.

I mean why is it necessary to manually persist file paths in Minio with empty files. File paths don't actually exist, they're just long file names with `/` in them.

Reply:
You're talking about logging in to Minio?

Reply:
yea - each users would get their own policy + secret key and access key.